src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCM.java

```diff
@@ -297,12 +297,15 @@ public FormValidation doCheckRepositorySlug(@QueryParameter String value) {
         @POST
         public FormValidation doCheckServerId(@QueryParameter String serverId) {
             Jenkins.get().checkPermission(Permission.CONFIGURE);
-            List<BitbucketServerConfiguration> serverList =
-                    new BitbucketPluginConfiguration().getServerList();
+            BitbucketPluginConfiguration pluginConfiguration = new BitbucketPluginConfiguration();
+            List<BitbucketServerConfiguration> serverList = pluginConfiguration.getValidServerList();
             // Users can only demur in providing a server name if none are available to select
-            if (!serverList.isEmpty() && isBlank(serverId)) {
+            if (serverList.stream().noneMatch(server -> server.getId().equals(serverId))) {
                 return FormValidation.error("Please specify a valid Bitbucket Server.");
             }
+            if (pluginConfiguration.hasAnyInvalidConfiguration()) {
```

Comment:
I feel like this check should go first. If the plugin configuration is invalid, then return an error.
Then check that the server ID matches one of the configured servers.
We won't need to `getValidServerList` if we do the server ID check after the invalid config check becuase we know that there are no invalid configurations. So it would probably mean we can get rid of that method too

Reply:
Do you think the user should be blocked from selecting a valid server configuration if any other invalid ones are present? The purpose of this signifier is to explain why servers the user has configured do not appear in the list given we no longer provide an explicit error when submitting incorrect feedback.